src/main/resources/META-INF/resources/primefaces/core/core.ajax.js

```diff
@@ -538,6 +538,12 @@ if (!PrimeFaces.ajax) {
 
                 var jqXhr = $.ajax(xhrOptions)
                     .fail(function(xhr, status, errorThrown) {
+                        if (xhr.status == 401) {
+                            var location = xhr.getResponseHeader("Location");
+                            PrimeFaces.debug('Unauthorized status received. Redirecting to ' + location);
+                            window.location = location;
```

Comment:
What if location is NULL?  Shouldn't it then just call `cfg.onerror.call` if no location URL is found?

Reply:
```java
if (xhr.status == 401) {
```

becomes

```java
var location = xhr.getResponseHeader("Location");
if (xhr.status == 401 && locataion) {
```

this solves the null location header problem previously mentioned.